feat: new squad AB variants

rebelchris opened the pull request:
## Changes

### Describe what this PR does
- Added new squads menu AB variants

Off:
Doesn't show at all

V1:
<img width="284" alt="Screenshot 2022-11-11 at 13 55 45" src="https://user-images.githubusercontent.com/554874/201335521-b701efff-c490-4385-9d58-323c3c41cedf.png">
<img width="181" alt="Screenshot 2022-11-11 at 13 55 51" src="https://user-images.githubusercontent.com/554874/201335532-323a27ee-514d-46a4-815f-aa13ed50e8de.png">

V2:
<img width="260" alt="Screenshot 2022-11-11 at 13 56 30" src="https://user-images.githubusercontent.com/554874/201335655-b273fa93-b604-4c85-9394-4a9ea01a7ea4.png">
<img width="167" alt="Screenshot 2022-11-11 at 13 56 34" src="https://user-images.githubusercontent.com/554874/201335660-1bd79490-90d2-464f-b6ed-e339f96fcf5f.png">

V3:
<img width="267" alt="Screenshot 2022-11-11 at 13 57 04" src="https://user-images.githubusercontent.com/554874/201335727-f5412df8-189b-4fd6-98d3-538f300e8de9.png">
<img width="184" alt="Screenshot 2022-11-11 at 13 57 10" src="https://user-images.githubusercontent.com/554874/201335734-fb5bf67c-90fb-4a30-b1a0-b781503a38e5.png">

V4:
<img width="281" alt="Screenshot 2022-11-11 at 13 57 39" src="https://user-images.githubusercontent.com/554874/201335812-be816cfa-0db7-4efa-860f-22a7bd83bec1.png">
<img width="200" alt="Screenshot 2022-11-11 at 13 57 44" src="https://user-images.githubusercontent.com/554874/201335819-6e3f1177-baf3-4323-aab2-247b448d3580.png">

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### **Please make sure existing components are not breaking/affected by this PR**

## Manual Testing

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

rebelchris commented:
Question for @idoshamun :
Can we do the logged users only via flagsmith or do we need to catch it on the frontend side?

And still awaiting if we need any tracking for this 🙏

el commented:
Looks great except for 2 things 
1- should it overlap?
![image](https://user-images.githubusercontent.com/1488164/201347767-4e4e9475-2260-4679-821a-5c0c2f2090a4.jpeg)

2- shouldn't be squads as in plural
![image](https://user-images.githubusercontent.com/1488164/201347989-0e24df6d-8016-4656-a243-0965f5d16df4.jpeg)

sshanzel reviewed (approved):
Spotless! ✨

rebelchris commented:
@el It's according design spec, but let me double check

idoshamun commented:
@rebelchris we will have to do the filtering on the app. So you will need to display it only for registered users.
Tracking is also needed as described in the taxonomy

idoshamun commented on packages/shared/src/components/sidebar/Sidebar.tsx:
```diff
@@ -65,6 +79,22 @@ export default function Sidebar({
 
+  const trackSquadClicks = () => {
+    trackEvent({
+      event_name: 'click create squads',
```
Fixed according to my changes in the taxonomy.
Please also add `target_id` as the variation of the button

```suggestion
      event_name: 'click create squad',
```

idoshamun commented on packages/shared/src/components/sidebar/Sidebar.tsx:
```diff
@@ -74,6 +104,15 @@ export default function Sidebar({
+    if (squadVersion !== SquadVersion.Off) {
+      trackEvent({
+        event_name: 'impression',
+        target_type: 'squads',
```
Fixed according to my changes in the taxonomy.
Please also add target_id as the variation of the button

```suggestion
        target_type: 'create squad',
```

idoshamun commented on packages/shared/src/components/sidebar/SquadButton.tsx:
```diff
@@ -0,0 +1,75 @@
+  onSquadClick: () => void;
+}
+
+function SquadButton({
```
The link should open in a new tab

rebelchris replied:
Ah it was mentioned in the kickoff to have it in the same tab, but will adjust :)

idoshamun commented on packages/shared/src/contexts/FeaturesContext.tsx:
```diff
@@ -60,6 +63,12 @@ export const FeaturesContextProvider = ({
         flags,
       ) as ShareVersion,
       authVersion: getFeatureValue(Features.AuthenticationVersion, flags),
+      squadVersion: getFeatureValue(
```
What's the difference between `squadButton` and `squadVersion`?

idoshamun replied:
Oh I see now, it's the copy of the button. Do we experiment on that?

rebelchris replied:
It's mentioned on Slack as an idea so better be prepared for it even if we don't use it.

idoshamun reviewed (changes requested):
Please see my comments

rebelchris commented:
@idoshamun Made the following changes:

- Adjusted the tracking events
- Adjust styling for the sidebar top margin (new spec 40px)
- Added target blank flow
- Added userID to link
- Ensure only logged users see the flow

Let me know if the squadButton text is good to go and any other remarks are open?

idoshamun commented on packages/shared/src/components/sidebar/Sidebar.tsx:
```diff
@@ -74,6 +107,16 @@ export default function Sidebar({
   );
 
+  useEffect(() => {
+    if (squadVersion !== SquadVersion.Off) {
```
In case the user is not logged in, but squad is on it will get tracked.
My suggestion is either create a component specifically for this button with the proper tracking.
Or have a boolean here in the sidebar that represents whether the squad button should be visible or not

rebelchris replied:
I will go with the boolean value, as because of the placement we can't use one generic component.

idoshamun reviewed (changes requested):
One more change and please also see the latest changes in the taxonomy due to the information you provided about the button copy and all

idoshamun commented on packages/shared/src/components/sidebar/Sidebar.tsx:
```diff
@@ -74,6 +110,18 @@ export default function Sidebar({
 
+  useEffect(() => {
+    if (squadVersion !== SquadVersion.Off && squadVisible) {
+      trackEvent({
```
My only concern is that it will be triggered multiple times, but we can leave it as it is for now. I can deal with it

rebelchris replied:
Yeah was indeed in two minds about using effect like this, it is unlikely it would change in one session though from my tests.

idoshamun replied:
Ok then :)
